python/task_manager/worker_thread.py

```diff
@@ -57,6 +55,7 @@ def shut_down(self):
         """
         Shut down the thread and wait for it to exit before returning
         """
+        self._results_poller = None
```

Comment:
paranoia is never a bad thing :) Just curious if it was for a particular reason! :)